dlib/matrix/matrix_windows.h

```diff
@@ -0,0 +1,124 @@
+                                                                                                \
+        WindowSymmetry t;                                                                       \
+                                                                                                \
+        const static long cost = M::cost + 7;                                                   \
+        typedef type const_ret_type;                                                            \
+        const_ret_type apply(long r, long c) const                                              \
+        {                                                                                       \
+            return function<R>(std::size_t(r*this->m.nc()+c), (std::size_t)this->m.size(), t) * this->m(r,c); \
+        }                                                                                       \
+    };                                                                                          \
+                                                                                                \
+    template <typename EXP>                                                                     \
+    const matrix_op<op_##function<EXP> > function (                                             \
+        const matrix_exp<EXP>& m,                                                               \
+        WindowSymmetry type                                                                         \
+    )                                                                                           \
+    {                                                                                           \
+        using op = op_##function<EXP>;                                                          \
+        return matrix_op<op>(op(m.ref(), type));                                                \
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    DLIB_DEFINE_WINDOW_OP(hann)
+    DLIB_DEFINE_WINDOW_OP(blackman)
+    DLIB_DEFINE_WINDOW_OP(blackman_nuttall)
+    DLIB_DEFINE_WINDOW_OP(blackman_harris)
+    DLIB_DEFINE_WINDOW_OP(blackman_harris7)
+
+// ----------------------------------------------------------------------------------------
+
+    template <typename M>
+    struct op_kaiser : basic_op_m<M>
+    {
+        using type = typename M::type;
+        using R    = remove_complex_t<type>;
+
+        op_kaiser(const M& m_, beta_t beta_, WindowSymmetry type_) : basic_op_m<M>(m_), beta{beta_}, t{type_}
+        {DLIB_ASSERT(is_vector(m_), "matrix expression must be a vector");}
+
+        beta_t beta;
+        WindowSymmetry t;
+
+        const static long cost = M::cost + 7;
+        typedef type const_ret_type;
+        const_ret_type apply(long r, long c) const
+        {
+            return kaiser<R>(std::size_t(r*this->m.nc()+c), (std::size_t)this->m.size(), beta, t) * this->m(r,c);
```

Comment:
Did you get warnings when not using these `size_t` casts?  Those shouldn't be needed.  Like this ought to do the right thing:
```suggestion
            return kaiser<R>(r*this->m.nc()+c, this->m.size(), beta, t) * this->m(r,c);
```

Reply:
MSVC complained I think when turning warning into errors.